File: dotnet/src/Experimental/Process.Abstractions/Models/KernelProcessStateMetadata.cs
// Copyright (c) Microsoft. All rights reserved.

using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace Microsoft.SemanticKernel.Process.Models;

/// <summary>
/// Process state used for State Persistence serialization
/// </summary>
public sealed record class KernelProcessStateMetadata : KernelProcessStepStateMetadata
{
    /// <summary>
    /// Process State of Steps if provided
    /// </summary>
    [DataMember]
    [JsonPropertyName("stepsState")]
    public Dictionary<string, KernelProcessStepStateMetadata>? StepsState { get; set; } = null;
}

File: python/semantic_kernel/processes/kernel_process/kernel_process_state_metadata.py
# Copyright (c) Microsoft. All rights reserved.


from typing import ClassVar, Literal

from pydantic import Field

from semantic_kernel.processes.kernel_process.kernel_process_step_state_metadata import (
    KernelProcessStepStateMetadata,
    TState,
)


class KernelProcessStateMetadata(KernelProcessStepStateMetadata[TState]):
    """Process state used for State Persistence serialization."""

    type_: Literal["Process"] = Field("Process", alias="$type")
    steps_state: dict[str, "KernelProcessStateMetadata | KernelProcessStepStateMetadata"] = Field(
        default_factory=dict, alias="stepsState"
    )

    model_config: ClassVar = {
        "populate_by_name": True,
    }
